sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryIO.java

```diff
@@ -2100,13 +2103,17 @@ public TableReference getOrCreateTable(BigQueryOptions options, String tableSpec
       return tableReference;
     }
 
-    /** Writes the accumulated rows into BigQuery with streaming API. */
-    private void flushRows(Bigquery client, TableReference tableReference,
-        List<TableRow> tableRows, List<String> uniqueIds, BigQueryOptions options) {
+    /**
+     * Writes the accumulated rows into BigQuery with streaming API.
+     */
+    private void flushRows(TableReference tableReference,
+        List<TableRow> tableRows, List<String> uniqueIds, BigQueryOptions options)
+            throws InterruptedException {
       if (!tableRows.isEmpty()) {
         try {
-          BigQueryTableInserter inserter = new BigQueryTableInserter(client, options);
-          inserter.insertAll(tableReference, tableRows, uniqueIds, byteCountAggregator);
+          long totalBytes = bqServices.getDatasetService(options).insertAll(
+              tableReference, tableRows, uniqueIds);
+          byteCountAggregator.addValue(totalBytes);
```

Comment:
In terms of backward compatibility, I think that that feature (_BigQueryTableInserter_) was there and was working in production and now we can't see any alternative to it, so... not that great.

I can't use `BigQueryIO.Write` otherwise I would have done it. Mainly because of the nature of our dataflows; I'll try to explain the scenario:
We are receiving batch of Json events from pub sub in base64 algorithm. We have to store those into Bigquey tables, so we translate these records into `GenericRecord` and then into `TableRows` in a `processElement()` method. In this method we retrieve the table schema and the name of the table at runtime (from the header of the message) so we can't `output()` a `List<TableRow>` otherwise we lose the table name externally and the schema as well, that's why we can't use `BigQueryIO.Write` and we have to store these records within the `processElement()` method itself using the object `BigQueryTableInserter`.

If you have any other idea to apply so, it's welcomed.
Thank you